cmds/exp/ssh/utils_plan9.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build plan9
+// +build plan9
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+)
+
+var consctl *os.File
+
+func init() {
+	// We have to hold consctl open so we can mess with raw mode.
+	var err error
+	consctl, err = os.OpenFile("/dev/consctl", os.O_WRONLY, 0755)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// cleanup turns raw mode back off and closes consctl
+func cleanup() {
+	consctl.Write([]byte("rawoff"))
+	consctl.Close()
+}
+
+// raw enters raw mode
+func raw() (err error) {
+	_, err = consctl.Write([]byte("rawon"))
+	return
+}
+
+// cooked turns off raw mode
+func cooked() (err error) {
+	_, err = consctl.Write([]byte("rawoff"))
+	return
+}
+
+// readPassword prompts the user for a password.
+func readPassword() (string, error) {
+	fmt.Print("Password: ")
+	raw()
+	cons, err := os.OpenFile("/dev/cons", os.O_RDWR, 0755)
```

Comment:
I would almost argue you want to do the open first before all this but ... seems ok. The mode is a bit weird, but that's a problem with this API more than anything ...